otri/database/database_stream.py

```diff
@@ -29,19 +33,41 @@ def __iter__(self) -> DatabaseIterator:
         Returns:
             The iterator for this object, returns a __DatabaseIterator object
         '''
-        pass
+        raise NotImplementedError(
+            "DatabaseStream is an abstract class, please implement this method in a subclass"
+        )
+
+    def is_closed(self) -> bool:
+        '''
+        Defines if new data might be added to the stream.
+        '''
+        raise NotImplementedError(
+            "DatabaseStream is an abstract class, please implement this method in a subclass"
+        )
+
+    def close(self):
+        '''
+        Prevents the stream from getting new data, data contained can still be iterated.
+        '''
+        raise NotImplementedError(
+            "DatabaseStream is an abstract class, please implement this method in a subclass"
+        )
 
 
 class _PostgreSQLIterator(DatabaseIterator):
 
-    def __init__(self, cursor):
+    def __init__(self, cursor, parent):
         '''
         Parameters:
             cursor
                 A cursor on which the query to iterate through has already been executed without fail.
+            parent : PostgreSQLStream
+                The parent Stream. Will be closed when the stream is over and the cursor closed.
         '''
         super().__init__()
         self.__cursor = cursor
+        self.__parent = parent
+        self.__buffer = None
 
     def __next__(self):
```

Comment:
Maybe, but maybe we might want to return their id in the database to modify that row, so I think it would be best to return the whole row, but this way we need a filter that turns each row into a json dict